rfcs/2022-12-09-541-reload-api.md

```diff
@@ -0,0 +1,208 @@
+# RFC 541 - 2022-12-09 - Reload API
+
+This RFC proposes adding an HTTP API for Vector's configuration reloading
+process. This will allow an enterprise control plane to more directly control a
+running instance and provide a path for synchronous runtime feedback that
+doesn't currently exist.
+
+## Context
+
+- [RFC: Observability Pipelines Worker (internal)](https://docs.google.com/document/d/10c6KuzbUcEUJxBfzX3a2CQaLHGPW2zKlz-_U8kS2DmA)
+- [Observability Pipelines Worker Implementation (internal)](https://docs.google.com/document/d/1hXv0fO1nsMkZeBIgZAsGD6QB9NACYaioxQ1FoQz94LI)
+
+## Cross cutting concerns
+
+- [Reloading RFC](https://docs.google.com/document/d/1rmLwFrx_yAeLHa_SeEyJx0Op6Z0EHpK6_LVa39fnlok)
+- [Add reload subcommand](https://github.com/vectordotdev/vector/issues/541)
+- [Data Samples for OP Configuration Builder (internal)](https://docs.google.com/document/d/1L8PUzDr1G9C3ORQPvMab6lcj-mh18P1nDq5ecnPf4mE)
+
+## Scope
+
+### In scope
+
+- Integrating an API that can initiate a reload and return success or failure
+- Refactoring Vector's main application loop to better support such an API
+- Allowing configuration error messages to be passed back via the API
+
+### Out of scope
+
+- Changing the actual reloading semantics of Vector in any way
+- Implementing APIs for any other type of control actions
+
+## Pain
+
+Our enterprise control plane component needs to initate configuration reloads
+in Vector and report back success or failure. Currently, the only way to do
+that would be to manipulate the config files on disk, ensure Vector is watching
+for changes, and parse the log output to determine whether or not the reload
+succeeded. This would be difficult to implement and likely unreliable.
+
+## Proposal
```

Comment:
Given this is an API interface it'd be nice to see a specification of the request/response here.